prefixset_iter_test.go

```diff
@@ -20,6 +20,8 @@ func TestPrefixSetAll(t *testing.T) {
 		{pfxs("::0/127", "::0/128"), pfxs("::0/127", "::0/128")},
 		{pfxs("::0/126", "::0/127"), pfxs("::0/126", "::0/127")},
 		{pfxs("::0/1", "::0/128"), pfxs("::0/1", "::0/128")},
+		{pfxs("fc00::/7"), pfxs("fc00::/7")},
```

Comment:
Thank you for including a test cases that exercise the bug.

Is there a reason to use a prefix length of 7? I think `"8000::/1"` would fit in with the test suite better. Granted, the shape of the tree would be the same since it's compressed, but using a prefix length of 7 suggests to the reader that there's something special about that length. I would prefer we either change this to `"8000::/1"` or add `"8000::/1"` as a test case above this one. You could also use `"c000::/2"` to minimally capture the qualitative difference between "len == 1" and "len > 1."

Reply:
> Is there a reason to use a prefix length of 7?

No particular reason. It was the prefix that led to the discovery of this bug. Updated per your suggestion and cherry-picked your commit.